core/src/main/kotlin/api/ProductsRoute.kt

```diff
@@ -397,15 +397,21 @@ fun Route.products() = route("products/{productId}") {
 
             val productId = call.requireIdParameter("productId")
             val createOrtRun = call.receive<CreateOrtRun>()
-
             val userDisplayName = call.principal<OrtPrincipal>()?.let { principal ->
                 UserDisplayName(principal.getUserId(), principal.getUsername(), principal.getFullName())
             }
 
-            val repositoryIds = productService.getRepositoryIdsForProduct(productId)
+            val repositoryIds = if (createOrtRun.repositoryIds.isEmpty()) {
+                productService.getRepositoryIdsForProduct(productId)
+            } else {
+                val productRepoIds = productService.getRepositoryIdsForProduct(productId)
+                createOrtRun.repositoryIds.filter { it in productRepoIds }.takeIf { it.isNotEmpty() }
+                    ?:
+                    throw IllegalArgumentException("None of the provided repository IDs belong to product $productId")
```

Comment:
I wonder if it would be better to always fail if at least one of the provided repository IDs does not belong to the product, otherwise users might not notice their mistake if it is silently ignored.

Reply:
> I wonder if it would be better to always fail if at least one of the provided repository IDs does not belong to the product, otherwise users might not notice their mistake if it is silently ignored.

![Screenshot 2025-04-18 at 05 05 10](https://github.com/user-attachments/assets/e49cb8ff-3dcf-4a87-9e53-b61e4ade0383)